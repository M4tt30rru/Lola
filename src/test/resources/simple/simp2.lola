##Import "std.lola"
##Find(CandidatesSequence) of ( ##OneOrMore ##Expression(_exp) ##separator , ) -> ##Expression(_val),
##Find(Candidates) ##OneOrMore ##CandidatesSequence(_cand)
##Find when (##Expression(_filter)) ##Candidates otherwise ##Expression(_def);
 ##replace ##ForEach(_cands) ##("Arrays.asList(" + ",".join([e.name for e in _._exps]) + ").stream().allMatch(" + _filter.name + ")?" + _._val.name + ":")
  ##(_def);

Integer i = when (a -> a.isFine())
 of (x, y(), Z.z) -> 3,
 of (w) -> W(),
 otherwise null;
