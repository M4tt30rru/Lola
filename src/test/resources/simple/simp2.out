Integer i = Arrays.asList(x, y(), Z.z).stream().allMatch(a -> a.isFine()) ? 3 :
 Arrays.asList(w).stream().allMatch(a -> a.isFine()) ? W() : null;
